src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -0,0 +1,141 @@
+package seedu.address.model.lesson;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+
+/**
+ * Represents a Lesson in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public abstract class Lesson {
+
+    // Time fields
+    private final Date date;
+    private final Time startTime;
+    private final Time endTime;
+
+    // Data fields
+    private final Subject subject;
+    private final Set<Homework> homework = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Lesson(Date date, Time startTime, Time endTime, Subject subject, Set<Homework> homework) {
+        requireAllNonNull(date, startTime, endTime, homework);
+        this.date = date;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.subject = subject;
+        this.homework.addAll(homework);
+    }
+
+    public Date getDate() {
+        return date;
+    }
+
+    public Time getStartTime() {
+        return startTime;
+    }
+
+    public Time getEndTime() {
+        return endTime;
+    }
+
+    public Subject getSubject() {
+        return subject;
+    }
+
+    /**
+     * Returns an immutable homework set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Homework> getHomework() {
+        return Collections.unmodifiableSet(homework);
+    }
+
+    /**
+     * Check if the Lesson object is recurring.
+     *
+     * @return True if it is a recurring lesson, false otherwise.
+     */
+    public boolean isRecurring() {
+        return false;
+    }
+
+    /**
+     * Update the lesson date to the same day on the following week.
+     *
+     * @return newDate The date of the same day on the following week.
+     */
+    public Date updateDateWithWeek() {
+        Date newDate = new Date(getDate().getLocalDate().plusDays(7).toString());
```

Comment:
I think the logic is also a bit dense here too ><. Can split into smaller logic chunks